pkgs/development/python-modules/mautrix/default.nix

```diff
@@ -37,19 +39,20 @@ buildPythonPackage rec {
 
   build-system = [ setuptools ];
 
-  dependencies = [
-    aiohttp
-    attrs
-    yarl
-  ];
-
-  passthru.optional-dependencies = {
-    detect_mimetype = [ python-magic ];
-    encryption = [
+  dependencies =
+    [
+      aiohttp
+      attrs
+      yarl
+    ]
+    ++ lib.optionals withOlm [
       python-olm
       unpaddedbase64
       pycryptodome
```

Comment:
We can turn this into a real extra

https://github.com/mautrix/python/blob/v0.20.6/setup.py#L33

Reply:
Or is the idea that the `optional-dependencies` would still be there but the `withOlm` flag would gate the tests? That would mean that downstream consumers could forget to enable it and be using untested functionality, but I guess there’s unlikely to be new ones added.